apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -50,6 +52,52 @@
     this.logger.debug('Clock updated', now);
   };
 
+	LockScreenClockWidget.prototype.updateAlarm =
+	function() {
+    this.fetchAlarmData()
+    .then((alarmData) => {
+      var type = 'AM',
+          hour = parseInt(alarmData.hour, 10);
+
+      // Decide wheter to use AM or PM
+      if(hour > 12) {
+        type = 'PM';
+        hour = hour - 12;
+      }
+
+      var fullAlarmTime = hour + ':' + alarmData.minute + type;
+      this.resources.elements.alarmtime.textContent = fullAlarmTime;
+      this.resources.elements.alarm.classList.remove('no-alarms');
+      this.logger.debug('Alarm updated to :', fullAlarmTime);
+    }).catch(() => {
+      this.resources.elements.alarm.classList.add('no-alarms');
+    });
+	};
+
+  /*
+   * Gets the Alarm Data from the DataStore.
+   */
+  LockScreenClockWidget.prototype.fetchAlarmData = function() {
+    return new Promise((resolve, reject) => {
+      navigator.getDataStores('alarms')
+      .then((stores) => {
+        stores[0].getLength()
+        .then((len) => {
+          if(len > 0) {
```

Comment:
Please invert this as I mentioned:

    if (len === 0) {
      return;
    }

Since it's an only one-branch conditional statement.